pkg/smokescreen/conntrack/conn_tracker.go

```diff
@@ -13,10 +13,19 @@ import (
 	"golang.org/x/net/publicsuffix"
 )
 
+type TrackerInterface interface {
+	ReportConnectionSuccessRate() *ConnSuccessRateStats
+	RecordAttempt(string, bool)
+	MaybeIdleIn(time.Duration) time.Duration
+	NewInstrumentedConn(net.Conn, *logrus.Entry, string, string, string) *InstrumentedConn
+	NewInstrumentedConnWithTimeout(net.Conn, time.Duration, *logrus.Entry, string, string, string) *InstrumentedConn
+	Wg() *sync.WaitGroup
```

Comment:
Nonblocking: I wonder if you'd get slightly more intuitive ergonomics by exposing higher-level functions instead of exposing the `WaitGroup` directly. So for example you could have `TrackerInterface.Drain()` or similar and avoid the caller having to think about the mechanism that a tracker uses to implement its connection tracking.